### PR TITLE
Planthunt

### PR DESCRIPTION
**Pull request by Zephyr75**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntJoinLobbyActivity.java [hunk 1]
```diff
@@ -62,21 +62,8 @@ public void onClick(View v) {
                                 }
                                 if (Math.toIntExact((long) task2.getResult().getValue()) < Math.toIntExact((long) task3.getResult().getValue())){
                                     int temp = db.addUserToLobby(lobbyName, username);
+                                    handleJoinLobby(temp, lobbyName, username);
```
Comment (Antonio06188) [inline app/src/main/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntJoinLobbyActivity.java:65]: good modularity

File: app/src/androidTest/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntWaitLobbyActivityTest.java [hunk 2]
```diff
@@ -47,8 +42,44 @@ public void waitScreenIsCorrectlyLoaded() {
         intent.putExtra(PlanthuntCreateJoinLobbyActivity.HOST_TYPE, PlanthuntCreateJoinLobbyActivity.HOST);
 
         try (ActivityScenario<PlanthuntWaitLobbyActivity> scenario = ActivityScenario.launch(intent)) {
-            Espresso.onView(withId(R.id.planthuntWaitButton)).check(matches(isDisplayed()));
+            Espresso.onView(withId(R.id.planthuntCreateLobbyButton)).check(matches(isDisplayed()));
         }
```
Comment (Antonio06188) [inline app/src/androidTest/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntWaitLobbyActivityTest.java:46]: Why does you replace planthuntWaitButton by planthuntCreateLobbyButton? I don't understand it

File: app/src/androidTest/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntLobbyActivityTest.java [hunk 3]
```diff
@@ -80,4 +81,26 @@ public void perform(UiController uiController, View view) {
...
+        assertEquals(lobby.getNbrPlayers(), 0);
+        assertEquals(lobby.getPlayersReady(), 0);
+        assertEquals(lobby.getPlayersGone(), 0);
+        assertEquals(lobby.getPlayerScore1(), 0);
+        assertEquals(lobby.getPlayerScore2(), 0);
+        assertEquals(lobby.getPlayerScore3(), 0);
+        assertEquals(lobby.getPlayerUid1(), "host");
+        assertEquals(lobby.getPlayerUid2(), "");
+        assertEquals(lobby.getPlayerUid3(), "");
+    }
```
Comment (Antonio06188) [inline app/src/androidTest/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntLobbyActivityTest.java:97]: Use a constant String and don't hardcode String, Int

File: app/src/androidTest/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntJoinLobbyActivityTest.java [hunk 4]
```diff
@@ -97,4 +99,18 @@ public void perform(UiController uiController, View view) {
...
         Espresso.onView(withId(R.id.planthuntJoinLobbyButton)).check(matches(isDisplayed()));
     }
+
+    @Test
+    public void handleJoinCorrectlyWorks() {
+        PlanthuntJoinLobbyActivity temp = new PlanthuntJoinLobbyActivity();
+        temp.handleJoinLobby(2, "lobby", "player");
+        temp.handleJoinLobby(1, "lobby", "player");
+        temp.handleJoinLobby(0, "lobby", "player");
+    }
```
Comment (Antonio06188) [inline app/src/androidTest/java/ch/epfl/sdp/healthplay/planthunt/PlanthuntJoinLobbyActivityTest.java:109]: Create a constant String, and Int

**Review by Antonio06188 — CHANGES_REQUESTED** [review]
Good work, I love your work on the imageButton, read my comment

**Review by Tetardfurieux — APPROVED** [review]

**Review by Antonio06188 — APPROVED** [review]

**Pull request merged**